chia/wallet/wallet_state_manager.py

```diff
@@ -1221,6 +1221,10 @@ async def new_peak(self, peak: wallet_protocol.NewPeakWallet):
         for wallet_id, wallet in self.wallets.items():
             if wallet.type() == uint8(WalletType.POOLING_WALLET):
                 await wallet.new_peak(peak.height)
+        current_time = int(time.time())
+
+        if self.wallet_node.last_wallet_tx_resend_time < current_time - self.wallet_node.wallet_tx_resend_timeout_secs:
+            self.tx_pending_changed()
```

Comment:
Yea, there are classes of errors that should likely never retry because they will never actually succeed. I think that will need to wait for a follow-up PR. I think versions < 1.3 always retried the transactions regardless of error as well

Reply:
No, this will not resend transactions that have been rejected by that specific peer.

This is handled elsewhere in the code. I agree this is not clear. Bit tangled in here.

Reply:
This will not load the network with unnecessary transactions though, it's only sending them to the nodes you are connected to. These nodes will just ignore it efficiently if they already have it.